Update main.yml

bstrauss84 opened the pull request:
Added node cordone/drain tasks to iptables sample.

bstrauss84 committed:
Update main.yml

Added node cordone/drain tasks to iptables sample.

andykrohg commented:
lgtm! Before merging this, can we update the README to indicate:
* how to create a dedicated one-node machineset for this with a taint, and
* an updated `oc run` command to ensure the privileged pod gets scheduled on our tainted node

andykrohg commented:
Another thought - maybe we could build the node drain into a secondary step in an Ansible Workflow Template that requires approval? So order could be
1. Kill the pod, and remove user rolebindings/clusterrolebindings
2. Prompt in ansible workflow for manual approval to continue
3. If yes, drain the node. (This would require a separate playbook)

Thoughts @bstrauss84 @roller1187 ?

bstrauss84 committed:
Merge branch 'andykrohg:main' into patch-1

bstrauss84 committed:
Update main.yml

bstrauss84 committed:
Update main.yml

modifying to attempt the user confirmation prompt/ansible workflow strategy Andy suggested.

andykrohg commented on samples/iptables-privileged-pod/main.yml:
```diff
@@ -59,3 +59,21 @@
+      path: "/tmp/node_name.txt"
+      state: absent
+
+# TODO: VERIFY. I know for our demo we are only spinning up one pod, but I believe the k8s module iss returning a list, so I'm specifying the first result in the list just to be safe, but need to test this. #
```
Does this task make sense to have here? The temp file will get blown away due to the nature of Automation Controller running the jobs in ephemeral EEs

bstrauss84 replied:
I figured that was the case, but was hoping it still held onto it if part of a workflow.  I haven't had much luck with other methods of passing the parameter betwixt two plays.

bstrauss84 replied:
Actually... I kinda wanna try using "set_stats" (https://docs.ansible.com/automation-controller/latest/html/userguide/workflows.html).  Standby...

bstrauss84 committed:
Create drain_node.yml

Creating second "node drain" playbook for the workflow.

bstrauss84 committed:
Merge branch 'andykrohg:main' into patch-1

bstrauss84 committed:
Update main.yml

trying out "set_stats"...

bstrauss84 committed:
Update drain_node.yml

andykrohg commented:
lgtm
